pipelines/deploy-fbc-operator/0.1/deploy-fbc-operator.yaml

```diff
@@ -183,27 +221,37 @@ spec:
 
                 # Get registry and repo from BUNDLE_IMAGE
                 reg_and_repo=$(get_image_registry_and_repository "${BUNDLE_IMAGE}")
-                mapfile -t mirrors < <(jq -r --arg image "${reg_and_repo}" '.[$image][]' /tekton/home/mirror-map.txt)
-                echo "Mirrors for $reg_and_repo are:"
-                printf "%s\n" "${mirrors[@]}"
-
-                # Iterate over mirrors and try each one
-                for mirror in "${mirrors[@]}"; do
-                  echo "Attempting to use mirror ${mirror}"
-                  replaced_image=$(replace_image_pullspec "$BUNDLE_IMAGE" "$mirror")
-
-                  # Check if the mirror is accessible
-                  if ! bundle_out=$(skopeo inspect --no-tags --config "docker://${replaced_image}"); then
-                    echo "Mirror $replaced_image is inaccessible."
-                    continue
-                  fi
 
-                  echo "Replacing $BUNDLE_IMAGE with $replaced_image"
-                  break
-                done
+                # Check if there are any mirrors for the image
+                has_mirrors=$(jq -r --arg image "${reg_and_repo}" 'has($image)' /tekton/home/mirror-map.txt)
+
+                if [[ "$has_mirrors" == "true" ]]; then
+                  mapfile -t mirrors < <(jq -r --arg image "${reg_and_repo}" '.[$image][]' /tekton/home/mirror-map.txt)
+                  
+                  echo "Mirrors for $reg_and_repo are:"
+                  printf "%s\n" "${mirrors[@]}"
+
+                  # Iterate over mirrors and try each one
+                  for mirror in "${mirrors[@]}"; do
```

Comment:
Before iterating the mirrors, we should probably check if the pullspec is accessible as it is. In that case, we can skip replacement.

Reply:
@yashvardhannanavati I just realized that I'm already checking if the bundle image is pullable at the very beginning of the step:https://github.com/konflux-ci/tekton-integration-catalog/blob/765ab726e01c9518a9d96ba8bdfbc779b0312bfa/pipelines/deploy-fbc-operator/0.1/deploy-fbc-operator.yaml#L172

If that's the case, we exit 0 skipping the mirror replacement